### PR TITLE
Improved tests and handling for 1-Wire communication errors

### PR DESCRIPTION
**Pull request by elcojacobs**
I noticed that running the tests under valgrind was slow. Investigation led to a bug that caused the DS2408 and DS2413 to update every millisecond in the test, instead of every second like they would in production.

Fixing the test uncovered that the communication error handling could be improved. I made some changes that cause an immediate retry on a communication error. The slow PWM module also does an immediate retry now and backs of for 1s if that also fails. This will reduce 1-wire traffic and PWM updates for disconnected devices too.

### CODE REVIEW

File: lib/control/src/DS2413.cpp [hunk 1]
```diff
@@ -21,40 +21,49 @@
...
+    };
+    bool wasConnected = connected();
+    bool success = doUpdate();
+    if (wasConnected && !success) {
+        // fast retry on first disconnect
+        success = doUpdate();
     }
     connected(success);
-    return success;
+    return 1000; // every second if already disconnected or everything is fine
```
Comment (steersbob) [inline lib/control/src/DS2413.cpp:66]: The comment suggests this is not the only `return` statement.

File: lib/blocks/src/DS2413Block.cpp [hunk 2]
```diff
@@ -87,8 +87,7 @@ DS2413Block::write(const cbox::Payload& payload)
 cbox::update_t
 DS2413Block::updateHandler(cbox::update_t now)
 {
-    _device.update();
-    return next_update_1s(now);
+    return now + _device.update();
```
Comment (steersbob) [inline lib/blocks/src/DS2413Block.cpp:90]: SHould return a duration, not abs time of next update.

**Review by steersbob — CHANGES_REQUESTED** [review]

File: lib/control/src/DS2413.cpp (hunk 1 above)
Comment (elcojacobs) [reply lib/control/src/DS2413.cpp:66]: It was not at some point in the refactor :). Well spotted.

File: lib/blocks/src/DS2413Block.cpp (hunk 2 above)
Comment (elcojacobs) [reply lib/blocks/src/DS2413Block.cpp:90]: actually it should return an absolute time. That was what next_update_1s did too.

**Review by steersbob — APPROVED** [review]

**Pull request merged**